### DISCUSSION TITLE
Porting TinyUF2 on NXP LPC55S36

### DISCUSSION THREAD

**Discussion by focus-martin** [adafruit/tinyuf2#240] [Q&A] [answered]
Hi everyone!

I am getting started with TinyUF2 and I have a couple of questions. My goal is to port TinyUF2 on NXP LPC55S36. I have already ported TinyUSB which is running well. Now I want to use TinyUF2 inside this MCU.

Here are my questions:

1. I took as reference the files located in ports/lpc55/boards/lpcxpresso55s28 and 55s69. My doubt about is how to define the `BOARD_FLASH_SIZE` macro for the MCU.
2. How the bootloader source code works?  My doubt is related to which is the source code that is executed by the bootloader. Is it the `main. c` inside the `src` folder, or the code inside the `app` folder?

Thanks in advance!

**Comment by gsteiert** [comment, marked as answer]
1.   In the existing LPC55 implementations, `BOARD_FLASH_SIZE` is set in the board.h file in the board directory.  
2.   main.c in the src folder is the bootloader.  The code in the app folder are for separate applications that can be generated to allow for performing common tasks like erasing the device or updating the bootloader.
